### PR TITLE
packets2: Add last-will-related packets

### PR DESCRIPTION
**Pull request by mprymek**
PR adds all last-will-related packets.

### CODE REVIEW

File: packets2/willmsgresp_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
+)
+
+func TestWillMsgRespConstructor(t *testing.T) {
+	assert := assert.New(t)
+
+	pkt := NewWillMsgResp(RC_SUCCESS)
```
Comment (dmajda) [inline packets2/willmsgresp_test.go:16]: Any reason why the code is changed from `RC_COGNESTION` used in the same test in `packets1`?

(Not that it matters. I just noticed the difference in the `packets1` ↔︎ `packets2` diff.)

File: packets2/willmsgresp_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	if pkt == nil {
+		t.Fatal("New packet should not be nil")
+	}
+
+	assert.Equal("*packets2.WillMsgResp", reflect.TypeOf(pkt).String(), "Type should be WillMsgResp")
+	assert.Equal(RC_SUCCESS, pkt.ReasonCode, "Default ReasonCode should be RC_ACCEPTED")
```
Comment (dmajda) [inline packets2/willmsgresp_test.go:23]: The message mentions a different code than the one passed as the expected value.

File: packets2/willtopic.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		WillTopic: willTopic,
+	}
+	p.computeLength()
+	return p
+}
+
+func (p *WillTopic) computeLength() {
+	// An empty WILLTOPIC message is a WILLTOPIC message without Flags and
+	// WillTopic field (i.e. it is exactly 2 bytes long).
+	// [MQTT-SN 2.0 specification draft WD19, chapter 3.1.8.3 Will Topic]
```
Comment (dmajda) [inline packets2/willtopic.go:36]: Sync to WD20?

File: packets2/willtopicresp_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
+)
+
+func TestWillTopicRespConstructor(t *testing.T) {
+	assert := assert.New(t)
+
+	pkt := NewWillTopicResp(RC_SUCCESS)
```
Comment (dmajda) [inline packets2/willtopicresp_test.go:16]: Same question as in `WillMsgTest`: Any reason why the code is changed from `RC_COGNESTION` used in the same test in `packets1`?

File: packets2/willtopicupd_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	assert.Equal(retain, pkt.Retain, "Bad Retain flag value")
+	assert.Equal(willTopic, pkt.WillTopic, "Bad WillTopic value")
+}
+
+func TestWillTopicUpdMarshal(t *testing.T) {
+	pkt1 := NewWillTopicUpd("test-topic", 1, true)
+	pkt2 := testPacketMarshal(t, pkt1)
+	assert.Equal(t, pkt1, pkt2.(*WillTopicUpd))
+
+	// Packet with zero-length topic is a special case (does not contain Flags).
```
Comment (dmajda) [inline packets2/willtopicupd_test.go:36]: „with **a** zero-length topic“?

**Review by dmajda — CHANGES_REQUESTED** [review]
Looks good, my comments are just trivialities.

File: packets2/willmsgresp_test.go (hunk 1 above)
Comment (mprymek) [reply packets2/willmsgresp_test.go:16]: Good point! The code in `packets1` was improved and I forgot to make the same improvement in `packets2`. `RC_SUCCES` is `0` which is Go's zero value => it's better to test the constructor with any other value.

Fixed.

File: packets2/willmsgresp_test.go (hunk 2 above)
Comment (mprymek) [reply packets2/willmsgresp_test.go:23]: Fixed.

File: packets2/willtopicresp_test.go (hunk 4 above)
Comment (mprymek) [reply packets2/willtopicresp_test.go:16]: Same as above. Fixed.

File: packets2/willtopicupd_test.go (hunk 5 above)
Comment (mprymek) [reply packets2/willtopicupd_test.go:36]: Fixed.

File: packets2/willtopic.go (hunk 3 above)
Comment (mprymek) [reply packets2/willtopic.go:36]: Done.

**Review by dmajda — APPROVED** [review]
Looks good now.

**Comment by mprymek** [comment]
For record: the previous force-push was just a rebase on top of the current `mqtt-sn-v2` branch HEAD.

**Pull request merged**